### PR TITLE
Sync parameter events callbacks

### PR DESCRIPTION
**Pull request by esteve**
Adds support for `on_parameter_event` to the `SyncParametersClient`

Connects to ros2/rclcpp#49

### CODE REVIEW

**Comment by esteve** [comment]
Do not review yet.

**Comment by esteve** [comment]
This is now ready for review.

**Comment by esteve** [comment]
Linux buildfarm:

http://ci.ros2.org/job/ros2_batch_ci_linux/10/

File: rclcpp/include/rclcpp/parameter_client.hpp [hunk 1]
```diff
@@ -308,6 +308,14 @@ class SyncParametersClient
     return rclcpp::executors::spin_node_until_future_complete(*executor_, node_, f).get();
   }
 
+  template<typename FunctorT>
+  typename rclcpp::subscription::Subscription<rcl_interfaces::msg::ParameterEvent>::SharedPtr
+  on_parameter_event(FunctorT & callback)
+  {
+    // TODO(esteve): remove hardcoded values
```
Comment (wjwwood) [inline rclcpp/include/rclcpp/parameter_client.hpp]: What does this comment mean?

**Comment by wjwwood** [comment]
lgtm other than the comment about the comment.

File: rclcpp/include/rclcpp/parameter_client.hpp (hunk 1 above)
Comment (esteve) [inline rclcpp/include/rclcpp/parameter_client.hpp]: Byproduct of copy/paste. Fixed.

**Comment by dirk-thomas** [comment]
lgtm

**Comment by tfoote** [comment]
+1

**Comment by jacquelinekay** [comment]
+1

**Pull request merged**